src/main/java/io/vertx/spi/cluster/zookeeper/impl/SubsMapHelper.java

```diff
@@ -162,4 +193,8 @@ public void childEvent(CuratorFramework client, TreeCacheEvent event) {
     }
   }
 
+  private void fireRegistrationUpdateEvent(String address, RegistrationInfo registrationInfo) {
+    nodeSelector.registrationsUpdated(new RegistrationUpdateEvent(address, Collections.singletonList(registrationInfo)));
```

Comment:
This is not right. The `registrationsUpdated` event should be fired with the latest view of all registration data for a specific address.

Reply:
Updated to use all the `RegistrationInfo` available using `get(address)`.